docs/math/eulerian.md

```diff
@@ -0,0 +1,68 @@
+在计算组合中，**欧拉数**（Eulerian Number）是从 $1$ 到 $n$ 中正好满足 $m$ 个元素大于前一个元素（具有 $m$ 个“上升”的排列）条件的排列 **个数** 。定义为：
+
+$$
+A(n, m) = 
+\langle 
+\begin{aligned}
+& n \\
+& m - 1 \\
+\end{aligned}
+\rangle
+$$
+
+例如，从数字 $1$ 到 $3$ 一共有 $4$ 种排列使得恰好有一个元素比前面的数字大：
+
+| 排列  |  满足要求的排列  | 个数  |
+| ---------- | --------------- | -------------- |
+| 1 2 3 | 1, 2 & 2, 3 | 2 |
+| 1 3 2 | 1, 3 | 1 |
+| 2 1 3 | 1, 3 | 1 |
+| 2 3 1 | 2, 3 | 1 | 
+| 3 1 2 | 1, 2 | 1 |
+| 3 2 1 |      | 0 |
+
+所以按照 $A(n, m)$ 定义： 如果 $n$ 等于 $3$ , $m$ 等于 $1$ ，欧拉数值为 $4$ ， 表示共有 $4$ 个有 $1$ 个元素大于前一个元素的排列。
+
+对于 $n$ 和 $m$ 值比较小的欧拉数来说，我们可以直接得到结果：
+
+| $A(n, m)$  | 满足要求的排列 | 个数 | 
+| ---------- | --------------- | --------------- | 
+| $A(1, 0)$ | $(1)$ | 1 |
+| $A(2, 0)$ | $(2, 1)$ | 1 |
+| $A(2, 1)$  | $(1, 2)$ | 1 |
+| $A(3, 0)$ | $(3, 2, 1)$ | 1 | 
+| $A(3, 1)$ | $(1, 3, 2), (2, 1, 3), (2, 3, 1), (3, 1, 2)$ | 4 |
+| $A(3, 2)$ | $(1, 2, 3)$ | 1 |
+
+## 公式
+
+$$
+A(n, m) = 
+\left\{
+\begin{array}{ll}
+0 & m > n \text{ or } n = 0 \\
+1 & m = 0 \\
+(n-m)*A(n-1, m-1) + (m+1) * A(n-1, m) & \text{otherwise} \\
```

Comment:
1. 建议挂一下参考文献
2. 稍微有点简陋了，如上面那个好兄弟说的，希望能添加一些类似于公式推导之类的。
3. 在不会引起歧义的情况下，请用 $\times$ 代替星号，叉乘请使用 $\times$，点乘请使用 $\cdot$。这里也可以直接留空，中间啥都不写。
您可以在下面这几个里挑一个好看的
```suggestion
(n-m)A(n-1, m-1) + (m+1)A(n-1, m) & \text{otherwise} \\
```
```suggestion
(n-m) \times A(n-1, m-1) + (m+1) \times A(n-1, m) & \text{otherwise} \\
```
```suggestion
(n-m) \cdot A(n-1, m-1) + (m+1) \cdot A(n-1, m) & \text{otherwise} \\
```
辛苦了.

Reply:
> 1. 建议挂一下参考文献
> 2. 稍微有点简陋了，如上面那个好兄弟说的，希望能添加一些类似于公式推导之类的。
> 3. 在不会引起歧义的情况下，请用 $\times$ 代替星号，叉乘请使用 $\times$，点乘请使用 $\cdot$。这里也可以直接留空，中间啥都不写。
>    您可以在下面这几个里挑一个好看的
> 
> 辛苦了.

好的，多谢修正！证明正在写了 :+1: